### PR TITLE
Remove gateway couch views

### PR DESCRIPTION
**Pull request by gcapalbo**
@esoergel 
@dannyroberts fyi

the only new commit here is the one to delete these views:

sms/backend_by_domain
sms/backend_by_owner_domain
sms/backend_map
sms/global_backends
sms/old_mobile_backend
telerivet/backend_by_secret

if we can get https://github.com/dimagi/commcare-hq/pull/10057 and https://github.com/dimagi/commcare-hq/pull/10100 merged before today's (Thursday's) deploy, i think we'll be in good shape if we wanted to make this be our Friday preindex. once those get merged i'll merge this with master to clean up the diff

### CODE REVIEW

File: corehq/pillows/mappings/domain_mapping.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from corehq.util.elastic import es_index
 
-DOMAIN_INDEX = es_index("hqdomains_20151204_1053")
```
Comment (esoergel) [inline corehq/pillows/mappings/domain_mapping.py]: You also need to update the test cases which have hardcoded references to this index name - they're in this file: https://github.com/dimagi/commcare-hq/blob/master/testapps/test_pillowtop/tests/data/all-pillow-meta.json#L160

**Comment by dannyroberts** [comment]
:x: (probably just the pointer ethan gave you)

Also, we should make sure and do this along because it looks like a big one. If I understand correctly, `_design/sms` currently has 16 views and you're removing 5 of them, so this is going to reindex 11 views, some of them over all message logs.

I *think* that should be fine, but I'm just flagging we should keep an eye on it and accept that it might bleed over into Monday and affect performance.

**Comment by gcapalbo** [comment]
build is failing on master: https://github.com/dimagi/commcare-hq/pull/10117

**Comment by czue** [comment]
@gcapalbo @dannyroberts should we not split up the rest of the `sms` design doc into `couchapps` so that we don't run into this again?

**Comment by gcapalbo** [comment]
@czue i'm not sure - would today's preindex take longer if there are 11 apps with 1 view each vs. 1 app with 11 views?  also note that the next time we have to reindex this app it will only have 7 views

**Comment by gcapalbo** [comment]
(or maybe splitting into 3 apps would perform the best?)

**Comment by czue** [comment]
I think today's preindex is the same, but the idea is that next time we need to add/remove/edit a view we only have to rebuild the one

**Comment by gcapalbo** [comment]
ok, i looked at the views and pretty much all that's left are views for logs and phone numbers. i might split it into two apps then (keep the logs in the sms app, and create a couchapp for the phone number views), since the next time we have to reindex we'll just be removing one or the other.

does that make sense or were you thinking 1 couchapp per view?

**Comment by czue** [comment]
that sounds fine to me if you expect it to all happen in bulk

**Comment by gcapalbo** [comment]
updated

**Comment by gcapalbo** [comment]
@dannyroberts this pr also removes an entire design doc (the one for telerivet). locally when i ran migrate it didn't remove it. is there anything special we need to do when we release this to remove it?

**Comment by czue** [comment]
@gcapalbo see [`prune_couch_views`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqadmin/management/commands/prune_couch_views.py)

**Comment by dannyroberts** [comment]
@gcapalbo splitting it up just into the chunks of views that you believe we'll delete together sounds like the right level of effort (one design doc per view would be a mess to deal with and wouldn't get us anything else), so this looks perfect.

**Comment by gcapalbo** [comment]
cool, then we're good to go if we wanted to preindex this one today.  does running prune_couch_views cause any indexing to happen, and at what point do we run that (we run it after we've deployed with these new view changes)?

**Comment by dannyroberts** [comment]
You can run it at any point in the future after the change that deletes a design doc has been deployed. It makes a list of design documents it expects to be there and subtracts that from the design docs that are there to make a list of design docs that it believes can be deleted with no consequence; it then asks you to confirm that list before deleting them. Because deleting an entire design doc doesn't require any reindexing, neither does `prune_couch_views`. Hope that helps! FWIW, I run this about once every week or two so it's no big deal if you don't remember to.

**Comment by dannyroberts** [comment]
And I agree, let's start this today.

**Comment by czue** [comment]
:boom:

**Pull request merged**